docs/resources/bunny_hostname.md

```diff
@@ -31,4 +32,12 @@ description: |-
 - **has_certificate** (Boolean) Determines if the hostname has an SSL certificate configured.
 - **is_system_hostname** (Boolean) Determines if this is a system hostname controlled by bunny.net.
 
+<a id="nestedblock--certificate"></a>
+### Nested Schema for `certificate`
+
+Required:
+
+- **certificate_data** (String) The public key.
```

Comment:
> - including CA chain.

Do I understand it right, that it means that all public-keys in the CA chain must be part of `certificate_data`?
Is that a hard requirement? I would expect that public keys of well-known CA can be missing, because they normally ship with the OS or browser of clients.

Reply:
yes. We have a "chain" entry for that in vault. its a set of certs bound together

Reply:
Maybe we should name there `certificate_crt` and `certificate_key` ?

Reply:
What is called here "public key" here must be a x509 pem encoded certificate + CA certificates. So it must contain what we have on vault under "crt" + what we have as "chain". Regarding the naming, I think `certificate` is more correct. For the certificate to work it must either have a `CN=domainmame` or have the domain names listed as part of the certificate's `SAN`.

The key must be a non-encrypted PEM key.

Reply:
> I would expect that public keys of well-known CA can be missing, because they normally ship with the OS or browser of clients.

Root CAs are expected to always be present on OSs, but not necessarily intermediary CAs. It is possible and common for certificates to be issued by an intermediary CA that is not known on some clients, and in this case the client can only validate the certificate if it is included on the server's certificate. As long as the client trusts the root CA, the browser will then accept the whole chain.

> Are you sure that only PEM encoding is supported on bunny side? I can't find any info on their page regarding supported encoding of the cert.

I'm sure it works with PEM encoding because it's the only I tested with. I also found no information on Bunny's site, other than the fact that is has to be then base64-encoded. The bunny go library already does the base64 encoding due to the behavior of go's `json` package, that encodes `[]byte` with base64.

Example of valid inputs for bunny:

For the Key:
```
-----BEGIN PRIVATE KEY-----
[private key here]
-----END PRIVATE KEY-----
```

For Certificate:
```
-----BEGIN CERTIFICATE-----
[certificate with CN=domainname or SAN]
-----END CERTIFICATE-----
-----BEGIN CERTIFICATE-----
[ca cert signing the cert above]
-----END CERTIFICATE-----
[other certificates on the chain]
```